gax-grpc/src/main/java/com/google/api/gax/grpc/OperationTimedPollAlgorithm.java

```diff
@@ -59,8 +59,19 @@ public static OperationTimedPollAlgorithm create(RetrySettings globalSettings) {
     super(globalSettings, clock);
   }
 
+  /**
+   * Returns {@code true} if another poll operation should be made or throws {@link
+   * CancellationException} otherwise.
```

Comment:
I'm probably missing a lot of context here, but why? Do we expect someone to catch the cancellation?

Reply:
It is catched in [BasicRetryingFuture](https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/retrying/BasicRetryingFuture.java#L163), allowing the retrying algorithm to cancel the future (there is a difference between failed future and a cancelled future).